src/ConstructiveSolidGeometry/VolumePrimitives/Ellipsoid.jl

```diff
@@ -47,13 +47,32 @@ This is a sphere with inner radius 1 and outer radius 2.
 
 See also [Constructive Solid Geometry (CSG)](@ref).
 """
-@with_kw struct Ellipsoid{T,CO,TR,TP,TT} <: AbstractVolumePrimitive{T, CO}
-    r::TR = 1
-    φ::TP = nothing
-    θ::TT = nothing
+struct Ellipsoid{T,CO,TR,TP,TT} <: AbstractVolumePrimitive{T, CO}
+    r::TR 
+    φ::TP 
+    θ::TT 
 
-    origin::CartesianPoint{T} = zero(CartesianPoint{T})
-    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
+    origin::CartesianPoint{T}
+    rotation::SMatrix{3,3,T,9}
+    
+    function Ellipsoid{T,CO}(r::TR,φ::TP,θ::TT,origin::CartesianPoint{T}=zero(CartesianPoint{T}),
+        rotation::SMatrix{3,3,T,9}=one(SMatrix{3, 3, T, 9})) where {T,CO,TR<:T,TP<:Nothing,TT<:Nothing} 
+        new{T,CO,TR,TP,TT}(r,φ,θ,origin,rotation)
+    end
+end
+
+
+function Ellipsoid(::Type{T}=Float64; r = 1.0, φ = nothing, θ = nothing, origin = zero(CartesianPoint{T}), 
```

Comment:
Right now, `φ` and `θ` can only be `nothing`. 
If we keep them here, there will be an error if they are not `nothing`.

We could expand this constructor with returning `CSGs` for non-full spheres, e.g. `CSGIntersection` of `Sphere` and `PartialTube` for partial `φ` or `CSGIntersection` of `Sphere` and `PartialCone` for partial `θ`.

Reply:
I guess we can leave out `φ` and `θ` in the constructors for now and just always force it to be nothing. 

These constructors should not return CSGs like `CSGDifference`, but always an `Ellipsoid`.